magma/ref.py

```diff
@@ -13,17 +13,34 @@ def __repr__(self):
 
 
 class AnonRef(Ref):
-    def __init__(self, name=""):
-        self.name = name
+    def __init__(self):
+        self.name = None
 
     def __str__(self):
-        return str(self.name) if self.name else f"AnonymousValue_{id(self)}"
+        return f"AnonymousValue_{id(self)}"
 
     def qualifiedname(self, sep='.'):
-        return str(self.name) if self.name else f"AnonymousValue_{id(self)}"
+        return f"AnonymousValue_{id(self)}"
+
+    def anon(self):
+        return True
+
+
+class NamedRef(Ref):
```

Comment:
I think we should call this `AnonNamedRef` or something...or inherit `DefnRef` and `InstRef` from `NamedRef`. It sounds like a baseclass (which it could be useful for).

Reply:
AnonNamedRef doesn't really make sense to me as a name (it's a contradiction), I'll see if I can weave it into a base class